### PR TITLE
MF-434 :  Make logo spot on navbar configurable

### PR DESCRIPTION
**Pull request by donaldkibet**
### What does this PR do?

1. Make the primary navigation navbar icon configurable.
### Screenshots

screencast 2021-04-07 11-00-44

### CODE REVIEW

File: src/components/logo/logo.component.tsx [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
-const Logo = ({ width = "110", height = "40" }) => {
+interface LogoProps {
+  src?: string;
+  alt?: string;
+  width?: number;
+  height?: number;
+}
+
+const Logo: React.FC<LogoProps> = ({
+  width = "110",
```
Comment (FlorianRappl) [inline src/components/logo/logo.component.tsx]: Thought this should be a `number`?

File: src/components/navbar/navbar.component.tsx [hunk 2]
```diff
@@ -88,7 +94,11 @@ const Navbar: React.FC<NavbarProps> = ({
                   hidePanel();
                 }}
               >
-                <Logo />
+                {config.logo.name ? (
+                  config.logo.name
+                ) : (
+                  <Logo src={config.logo?.src} alt={config.logo?.alt} />
```
Comment (FlorianRappl) [inline src/components/navbar/navbar.component.tsx]: Here we say logo may be undefined or null and above we assume its never null and check `config.logo.name`. Something is not right.

**Review by anjula-sack — APPROVED** [review]

**Comment by brandones** [comment]
@donaldkibet I'll let Florian finish reviewing, since he's already on it and I didn't read very closely, but: it's really nice to see someone do a good job of schema design on the first try. Great work.

**Review by tejasdeshpande100 — COMMENTED** [review]
LGTM👍🏻

File: src/components/logo/logo.component.tsx [hunk 3]
```diff
@@ -1,10 +1,26 @@
...
+
+const Logo: React.FC<LogoProps> = ({ width = 110, height = 40, config }) => {
+  const { logo } = config;
   return (
-    <svg role="img" width={width} height={height}>
-      <use xlinkHref="#omrs-logo-partial-grey"></use>
-    </svg>
+    <>
+      {logo?.src ? (
+        <img src={logo.src} alt={logo.alt} width={width} height={height} />
```
Comment (harshit151001) [inline src/components/logo/logo.component.tsx]: What if logo.alt is `undefined`, Why not use optional chaining here also?

**Review by harshit151001 — COMMENTED** [review]
What if `logo.alt` is `undefined`, Why not use optional chaining here also?

File: src/components/logo/logo.component.tsx (hunk 3 above)
Comment (tejasdeshpande100) [reply src/components/logo/logo.component.tsx]: Good suggestion @harshit151001. I guess we should consider displaying a default alternate text eg. "logo" if logo.alt is null/undefined.

Comment (brandones) [reply src/components/logo/logo.component.tsx]: `logo.alt` will never be undefined. "Logo" is the default provided by the config schema.

Comment (tejasdeshpande100) [reply src/components/logo/logo.component.tsx]: Oops, overlooked it!

File: src/components/logo/logo.component.tsx [hunk 4]
```diff
@@ -1,10 +1,26 @@
...
 import * as React from "react";
 
-const Logo = ({ width = "110", height = "40" }) => {
+interface LogoProps {
+  config: ConfigObject;
+  width?: number;
+  height?: number;
+}
+
+const Logo: React.FC<LogoProps> = ({ width = 110, height = 40, config }) => {
```
Comment (brandones) [inline src/components/logo/logo.component.tsx]: No need to pass `config` through as a prop. Eliminate it from the parent component and the props of this component and just do

```typescript
const { logo } = useConfig()
```

in this component.

Comment (brandones) [reply src/components/logo/logo.component.tsx]: Then to test, mock `useConfig`.

Comment (brandones) [reply src/components/logo/logo.component.tsx]: Also what's the point of having the width and height as props if they're not passed through by any parent component? This component doesn't need any props at all. Just inline the width and height where they're used. Keep it simple.

**Review by brandones — CHANGES_REQUESTED** [review]
Sorry, just remembered that Florian's on vacation this week. I'll review. Let's simplify things a bit. Generally looks good though.

File: src/components/logo/logo.component.tsx (hunk 4 above)
Comment (donaldkibet) [reply src/components/logo/logo.component.tsx]: made the above changes

**Review by Bawanthathilan — APPROVED** [review]

File: src/components/navbar/navbar.component.tsx [hunk 5]
```diff
@@ -34,6 +39,7 @@ const Navbar: React.FC<NavbarProps> = ({
   allowedLocales,
   session
 }) => {
+  const config = useConfig();
```
Comment (brandones) [inline src/components/navbar/navbar.component.tsx]:
```suggestion
```

File: src/components/navbar/navbar.component.tsx [hunk 6]
```diff
@@ -4,7 +4,12 @@ import UserMenuPanel from "../navbar-header-panels/user-menu-panel.component";
 import SideMenuPanel from "../navbar-header-panels/side-menu-panel.component";
 import Logo from "../logo/logo.component";
 import { isDesktop } from "../../utils";
-import { useLayoutType, navigate, ExtensionSlot } from "@openmrs/esm-framework";
+import {
+  useLayoutType,
+  navigate,
+  ExtensionSlot,
+  useConfig
```
Comment (brandones) [inline src/components/navbar/navbar.component.tsx]:
```suggestion
  ExtensionSlot
```

**Review by brandones — APPROVED** [review]
A couple cleanup suggestions, but looks good! Thanks so much @donaldkibet !

**Pull request merged**